src/tools/tracejob.c

```diff
@@ -481,13 +488,35 @@ parse_log(FILE *fp, char *job, int ind)
 
 			if (tmp.date != NULL) {
 				log_lines[ll_cur_amm].date = strdup(tmp.date);
-				if (sscanf(tmp.date, "%d/%d/%d %d:%d:%d", &tms.tm_mon, &tms.tm_mday, &tms.tm_year, &tms.tm_hour, &tms.tm_min, &tms.tm_sec) != 6)
-					log_lines[ll_cur_amm].date_time = -1;	/* error in date field */
-				else {
-					if (tms.tm_year > 1900)
-						tms.tm_year -= 1900;
-					tms.tm_mon--;         /* The number of months since January, in the range 0 to 11 for mktime */
-					log_lines[ll_cur_amm].date_time = mktime(&tms);
+				if ((ind != IND_ACCT) && (strchr(tmp.date, '.'))) {
+					if (sscanf(tmp.date, "%d/%d/%d %d:%d:%d.%ld", &tms.tm_mon, 
+					    &tms.tm_mday, &tms.tm_year, &tms.tm_hour, &tms.tm_min, 
+					    &tms.tm_sec, &(log_lines[ll_cur_amm].highres_ts)) != 7) {
+						log_lines[ll_cur_amm].date_time = -1;	/* error in date field */
+						log_lines[ll_cur_amm].highres_ts = -1;
+					} else {
+						has_high_res_timestamp = 1;
+						if (tms.tm_year > 1900)
+							tms.tm_year -= 1900;
+						/* The number of months since January, 
+ 						 * in the range 0 to 11 for mktime()
+ 						 */
+						tms.tm_mon--;
+						log_lines[ll_cur_amm].date_time = mktime(&tms);
```

Comment:
Don;t you need to set 'log_lines[ll_cur_amm].highres_ts' here? I don't know what it's used for, but it's being set in other cases, so just checking

Reply:
This is the case where highres_ts was set by the sscanf().  The other cases are either an error in the sscanf() or no microseconds were detected.  Those need to have it set to -1.